### PR TITLE
Updates healthcare shared components to 2.1.0.

### PR DESCRIPTION
**Pull request by rotodd**
## Description
Updates the healthcare shared components package to 2.1.0 to pull in changes that make SQL retry logic accessible. These changes will allow #1811 to build. 

## Related issues
N/A

## Testing
All tests should pass as before.

## FHIR Team Checklist
- ✅ **Update the title** of the PR to be succinct and less than 50 characters
- ✅ **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- ✅ Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- ✅ Tag the PR with **Azure API for FHIR** if this will release to the managed service
- ✅ Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
None (updates nuget)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Api/Features/ApiNotifications/ApiNotificationMiddleware.cs [hunk 1]
```diff
@@ -64,7 +64,7 @@ protected virtual async Task PublishNotificationAsync(HttpContext context, Reque
                 }
                 finally
                 {
-                    apiNotification.Latency = timer.GetElapsedTime();
+                    apiNotification.Latency = timer?.ElapsedTime ?? TimeSpan.Zero;
```
Comment (Ivanidzo4ka) [inline src/Microsoft.Health.Fhir.Api/Features/ApiNotifications/ApiNotificationMiddleware.cs:67]: >timer [](start = 46, length = 5)

[https://github.com/microsoft/healthcare-shared-components/blob/master/src/Microsoft.Health.Core/ActionTimerExtensions.cs](https://github.com/microsoft/healthcare-shared-components/blob/master/src/Microsoft.Health.Core/ActionTimerExtensions.cs)
Am I missing something, or it's for defensive purpose?

**Review by brendankowitz — APPROVED** [review]
🚢

File: src/Microsoft.Health.Fhir.Api/Features/ApiNotifications/ApiNotificationMiddleware.cs (hunk 1 above)
Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.Api/Features/ApiNotifications/ApiNotificationMiddleware.cs:67]: True, I wouldn't think `timer` would be null within the using scope

Comment (rotodd) [reply src/Microsoft.Health.Fhir.Api/Features/ApiNotifications/ApiNotificationMiddleware.cs:67]: With the package updates needed, it looks like `GetElapsedTime()` is now a property. The null check was just to get rid of the warning...

Comment (rotodd) [reply src/Microsoft.Health.Fhir.Api/Features/ApiNotifications/ApiNotificationMiddleware.cs]: Removed the null check in 29af3c4.

**Review by Ivanidzo4ka — APPROVED** [review]
:shipit:

**Pull request merged**